packages/core/src/external/athenahealth/index.ts

```diff
@@ -322,21 +376,93 @@ class AthenaHealthApi {
     return medicationOptions;
   }
 
+  async getAppointments({
+    cxId,
+    departmentId,
+    startAppointmentDate,
+    endAppointmentDate,
+    startLastModifiedDate,
+    endLastModifiedDate,
+  }: {
+    cxId: string;
+    departmentId: string;
+    startAppointmentDate: Date;
+    endAppointmentDate: Date;
+    startLastModifiedDate: Date;
+    endLastModifiedDate: Date;
+  }): Promise<{ appointments: AthenaAppointment[] }> {
+    const { log, debug } = out(`AthenaHealth get appointments`);
+    const params = {
+      departmentid: this.stripDepartmentId(departmentId),
+      startdate: this.formatDate(startAppointmentDate.toISOString()) ?? "",
+      enddate: this.formatDate(endAppointmentDate.toISOString()) ?? "",
+      startlastmodified: this.formatDateTime(startLastModifiedDate.toISOString()) ?? "",
+      endlastmodified: this.formatDateTime(endLastModifiedDate.toISOString()) ?? "",
+    };
+    const urlParams = new URLSearchParams(params);
+    try {
+      const appointmentUrl = `/appointments/booked?${urlParams.toString()}`;
+      const response = await this.handleAxiosRequest(() =>
+        this.axiosInstanceProprietary.get(appointmentUrl)
+      );
+      if (!response.data) throw new Error(`No body returned from ${appointmentUrl}`);
```

Comment:
hypernit: `nobody`

Reply:
again, referring to req body, not a person.